### ISSUE TITLE
How to update the server

### ISSUE THREAD

**Issue by MrJolly** [mschnitzer/ark-survival-ascended-linux-container-image#4]
Thanks for the share, the install went smooth and I've had a run around but the server files have updated online. How do I get the server to update in the docker ?

**Comment by mschnitzer** [comment]
Hi,

just run `docker restart asa-server`. It looks for an update on every container start.

**Comment by MrJolly** [comment]
That didn't seem to do anything but I will try again later, the update might not be available to unofficial servers just yet.

**Comment by mschnitzer** [comment]
Upon every server start, the startup script calls steam to validate the server files:
https://github.com/mschnitzer/ark-survival-ascended-linux-container-image/blob/main/root/usr/bin/start_server#L11

Steam would download missing files (or updated files) eventually.

According to steamdb (https://steamdb.info/app/2430930/depots/) the public branch of the server got updated ~2 days ago. So I speculate there's no update available at the moment, so you should be fine.

**Comment by MrJolly** [comment]
That's great, thanks. I was confused because the launcher says 25.61 is the latest but my server is 25.55. Evolved would sometimes take a while to update to the latest version.
In an unrelated issue, how would I give a file ( game.ini ) in the Saved folder the same 25000 permissions as the gamesettings.ini ?

**Comment by mschnitzer** [comment]
> That's great, thanks. I was confused because the launcher says 25.61 is the latest but my server is 25.55. Evolved would sometimes take a while to update to the latest version.

Yes, mine is also still on 25.55 while the client is ahead. Also they do release client-only updates from time to time, so that's not an issue at all. But I can promise that the server is updating just fine once an update has been released for the server as well.

> In an unrelated issue, how would I give a file ( game.ini ) in the Saved folder the same 25000 permissions as the gamesettings.ini ?

That's not needed. The reason for that is, that they `docker-compose.yml` file has a definition for a container called `set-permissions` (https://github.com/mschnitzer/ark-survival-ascended-linux-container-image/blob/main/docker-compose.yml#L22) of which the asa-server container depends on. This means upon every launch of the server via `docker-compose up -d` (must be with `docker-compose`, not just `docker restart`), the `set-permissions` container sets the permissions of all files to 25000:25000 automatically, so that you don't have to worry about it.

Anyway, if you still want to do this manually outside of the container, you can do so by running:

```
chown -R 25000:25000 /var/lib/docker/volumes/ark-asa-server-deployment_server-files/_data/*
```

That does the trick as well.

**Comment by MrJolly** [comment]
Brill, I think you answered all of my noob questions. I already tried to chown the files but user 25000 doesn't exist outside of the docker container. If running docker-compose up -d will do it I'll try that, thanks for taking the time to answer.

**Comment by mschnitzer** [comment]
Does you OS really complain about the user not existing? If so, which OS are you using?

Just asking because I'm running containers since years and my default uid and gid is 25000 (decision by me, not by docker) and I never create a user or group owning these ids. I'm just fine chowning with these ids.

**Comment by MrJolly** [comment]
Running Ubuntu Server 20.04, might be something unique to that.

**Comment by mschnitzer** [comment]
Can you paste the console output of Ubuntu complaining?

**Comment by MrJolly** [comment]
Ooh, that's strange. I just tried it again using webmin and it worked, thanks for the info.

**Comment by MrJolly** [comment]
Think you've sorted out all of my ( Ark ) issues, thanks for the top notch support.
